compiler/src/iree/compiler/DispatchCreation/SetSplitReductionSizes.cpp

```diff
@@ -239,6 +257,20 @@ struct SetSplitReductionSizesPass final
       return std::nullopt;
     }
 
+    // Compute the product of the specified dimensions. If any dimension list is
+    // empty, return 1.
+    auto getSizeAt = [](ArrayRef<int64_t> shape, ArrayRef<int64_t> pos) {
+      int64_t totalSize = 1;
+      for (unsigned i : pos)
+        totalSize *= shape[i];
```

Comment:
Please add an assert here that the shape[i] is not dynamic. This is relying on that condition being checked above, but if you drop that condition there it will mess up this logic. Also Nit, IREE norm is to always uses `{` `}` even if there is just one statement